app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/image_import_workflow.rb

```diff
@@ -20,87 +51,119 @@ def post_execute
       **options[:cos_pvs_creds]
     }
 
+    msg = nil
+    signal = nil
+    status = nil
+
     ems.with_provider_connection(:service => 'PCloudImagesApi') do |api|
       response = api.pcloud_cloudinstances_images_post(ems.uid_ems, body, {})
-      context[:task_id] = response.taskref.task_id
+      raise MiqException, 'HTTP response object is malformed, remote API error' if response&.taskref&.task_id.nil?
 
-      started_on = Time.now.utc
-      update!(:context => context, :started_on => started_on)
-      miq_task.update!(:started_on => started_on)
+      context[:task_id] = response.taskref.task_id
+      update!(:context => context)
+
+      msg = 'initiated OVA file importing from the cloud object storage'
+      signal = :post_execute_poll
+      status = 'ok'
+    rescue => e
+      msg = "failed to initiate OVA file importing from the cloud object storage: '#{trunc_err_msg(e.message)}'"
+      signal = :abort
+      status = 'error'
+      _log.error("OVA image import failure: '#{e.message}'")
     end
 
-    cleanup_git_repository
-
-    queue_signal(:post_execute_poll, 'importing image into PVS', 'running')
-  end
-
-  def post_execute_poll(*args)
-    msg, status = args
     set_status(msg, status)
+    queue_signal(signal, msg, status)
+  end
 
+  def post_execute_poll(*_args)
     ems = ExtManagementSystem.find(options[:ems_id])
 
     msg    = nil
     status = nil
     signal = nil
+    max_retries = 16
 
     ems.with_provider_connection(:service => 'PCloudTasksApi') do |api|
       response = api.pcloud_tasks_get(context[:task_id])
+      raise MiqException, 'HTTP response object is malformed, remote API error' if response&.status.nil?
```

Comment:
This seems like an odd thing to handle unless you actually hit it, when would the API call not return _any_ http status?

Reply:
Since we don't control the remote site and can't be completely sure if the response is well formed I though to guard against nullptr here. However, this is probably a job a response validator from the SDK client gem. I will remove.